src/scss/grommet-core/_objects.card.scss

```diff
@@ -53,6 +53,10 @@ $card-hover-color: #EBEBEB;
 }
 
 .#{$grommet-namespace}card__content {
+  .#{$grommet-namespace}tiles & {
+    flex-grow: 1;
```

Comment:
Updated the PR to apply `flex="grow"` on the card content box. It seems to work on Tiles (where we want vertical stretching within a row) without adversely affecting masonry Columns (vertical stretching not desired).